src/Language/DifferentialDatalog/OVSDB/Compile.hs

```diff
@@ -85,11 +88,15 @@ data TableKind = TableInput
                | TableDeltaUpdate
                deriving(Eq)
 
+-- | Output relation configuration:
+-- (name, Left <set_of_readonly_columns> or Right <Set of rw columns>)
+type OutputRelConfig = (String, Either [String] [String])
```

Comment:
Since you went to the trouble of defining a type you could probably have defined one with named fields; this would make it easier to read the code. But this is fine, since it's not used that much.